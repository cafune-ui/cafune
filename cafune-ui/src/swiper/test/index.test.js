import Swiper from '../';
const { SwiperItem } = Swiper;
import { render } from 'enzyme';
import { shallow, deep } from 'preact-render-spy';

describe('<Swiper />', () => {
  it('should render porperly', () => {
    const wrapper = render(
      <Swiper>
        {Array(...Array(4)).map((_, ind) => (
          <SwiperItem key={ind}>{ind + 1}</SwiperItem>
        ))}
      </Swiper>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a customIndicator', () => {
    const indi = (
      <div style="padding: 4px 10px;position: absolute;right: 10px;bottom:10px;background: rgba(0,0,0, .6);color: #fff;font-size:12px;">
        {2}/4
      </div>
    );
    const wrapper = deep(
      <Swiper customIndicator={indi}>
        {Array(...Array(4)).map((_, ind) => (
          <SwiperItem key={ind}>{ind + 1}</SwiperItem>
        ))}
      </Swiper>
    );
    expect(wrapper).toMatchSnapshot();

    wrapper.render(null);
    expect(wrapper).toMatchSnapshot();
  });
});
