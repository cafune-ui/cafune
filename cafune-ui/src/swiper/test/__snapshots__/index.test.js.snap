// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Swiper /> should render a customIndicator 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "customIndicator": VNode {
        "attributes": undefined,
        "children": Array [
          "2/4",
        ],
        "key": undefined,
        "nodeName": "div",
      },
      "duration": 200,
      "initialIndex": 3,
      "intervel": 500,
      "onChange": [MockFunction] {
        "calls": Array [
          Array [
            3,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "slideWidth": 280,
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "key": 0,
        },
        "children": Array [
          1,
        ],
        "key": 0,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "key": 1,
        },
        "children": Array [
          2,
        ],
        "key": 1,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "key": 2,
        },
        "children": Array [
          3,
        ],
        "key": 2,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "key": 3,
        },
        "children": Array [
          4,
        ],
        "key": 3,
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`<Swiper /> should render nothing 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "initialIndex": 0,
    },
    "children": Array [],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`<Swiper /> should render porperly 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "autoplay": true,
      "initialIndex": 0,
      "showIndicators": true,
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "key": 0,
        },
        "children": Array [
          1,
        ],
        "key": 0,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "key": 1,
        },
        "children": Array [
          2,
        ],
        "key": 1,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "key": 2,
        },
        "children": Array [
          3,
        ],
        "key": 2,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "key": 3,
        },
        "children": Array [
          4,
        ],
        "key": 3,
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;
